com.unity.ml-agents/Runtime/SideChannels/EnvironmentParametersChannel.cs

```diff
@@ -0,0 +1,90 @@
+using System.Collections.Generic;
+using System;
+
+namespace MLAgents.SideChannels
+{
+    /// <summary>
+    /// Lists the different data types supported.
+    /// </summary>
+    internal enum EnvironmentDataTypes
+    {
+        Float = 0
+    }
+
+    /// <summary>
+    /// A side channel that manages the environment parameter values from Python. Currently
+    /// limited to parameters of type float.
+    /// </summary>
+    internal class EnvironmentParametersChannel : SideChannel
+    {
+        Dictionary<string, float> m_Parameters = new Dictionary<string, float>();
+        Dictionary<string, Action<float>> m_RegisteredActions =
+            new Dictionary<string, Action<float>>();
+
+        const string k_EnvParamsId = "534c891e-810f-11ea-a9d0-822485860400";
+
+        /// <summary>
+        /// Initializes the side channel. The constructor is internal because only one instance is
+        /// supported at a time, and is created by the Academy.
+        /// </summary>
+        internal EnvironmentParametersChannel()
+        {
+            ChannelId = new Guid(k_EnvParamsId);
+        }
+
+        /// <inheritdoc/>
+        public override void OnMessageReceived(IncomingMessage msg)
+        {
+            var key = msg.ReadString();
+            var type = msg.ReadInt32();
+            if ((int)EnvironmentDataTypes.Float == type)
+            {
+                var value = msg.ReadFloat32();
+
+                m_Parameters[key] = value;
+
+                Action<float> action;
+                m_RegisteredActions.TryGetValue(key, out action);
+                action?.Invoke(value);
+            }
+            else
+            {
+                throw new UnityAgentsException("EnvironmentParametersChannel only supports floats.");
```

Comment:
We should either silently ignore unhandled types here, or at most log a warning (no exception).

Reply:
@vincentpierre